src/kbmod_wf/single_chip_workflow2.py

```diff
@@ -0,0 +1,197 @@
+import argparse
+import os
+import glob
+
+import toml
+import parsl
+from parsl import python_app, File
+import parsl.executors
+
+from kbmod_wf.utilities import (
+    apply_runtime_updates,
+    get_resource_config,
+    get_executors,
+    get_configured_logger,
+)
+
+from kbmod_wf.workflow_tasks import create_manifest, ic_to_wu, kbmod_search
+
+
+@python_app(
+    cache=True,
+    executors=get_executors(["local_dev_testing", "sharded_reproject"]),
+    ignore_for_cache=["logging_file"],
+)
+def step1(inputs=(), outputs=(), runtime_config={}, logging_file=None):
+    from kbmod_wf.utilities.logger_utilities import get_configured_logger, ErrorLogger
+    logger = get_configured_logger("kbmod.ic_to_wu")
+
+    import numpy as np
+    from reproject.mosaicking import find_optimal_celestial_wcs
+
+    from kbmod import ImageCollection
+    from kbmod.configuration import SearchConfiguration
+    import kbmod.reprojection as reprojection
+
+    from lsst.daf.butler import Butler
+    
+    logger.info(f"Running with config {runtime_config}")
+
+    with ErrorLogger(logger):
+        # Unravell inputs
+        repo_root = runtime_config["butler_config_filepath"]
+        search_conf_path = runtime_config.get("search_config_filepath", None)
+        ic_file = inputs[0].filepath
+
+        # Run core tasks
+        logger.info("Reading ImageCollection and adjusting search limits.")
+        ic = ImageCollection.read(ic_file)
+        ic.data.sort("mjd_mid")
+        butler = Butler(repo_root)
+        search_conf = SearchConfiguration.from_file(search_conf_path)
+
+        # fixup config to match specifics of the image collection in question
+        search_conf._params["n_obs"] = len(ic)/2
+        logger.info(f"Setting search config n_obs to {search_conf._params['n_obs']}")
+        
+        # Fit the optimal WCS
+        # TODO: triple check this doesn't flip the array, I'm pretty sure it does
+        opt_wcs, shape = find_optimal_celestial_wcs(list(ic.wcs))
+        opt_wcs.array_shape = shape
+
+        wu = ic.toWorkUnit(search_config=search_conf, butler=butler, overwrite=True)
```

Comment:
This line will create a sharded wu by default yeah? I can't seem to remember if we need to pass a `sharded=True`, or if the method is `toShardedWorkUnit` or if it just happens automatically.

Reply:
It won't shard by default, no. I figured that since parallelization won't benefit from separate IO (price already paid when standardizing images into work unit) so I figure one large write beats shared ones.

Reply:
Gotcha, that makes sense.